CommandDotNet/Help/HelpTextProvider.cs

```
@@ -206,8 +207,77 @@ protected virtual string SubcommandHelpHint(Command command) =>
+        {
+            var methodName = description.Substring("__DESCRIPTION_METHOD__".Length, 
+                description.Length - "__DESCRIPTION_METHOD__".Length - 2);
+            description = ResolveDescriptionMethod(argument, methodName);
+        }
+        
+        return description.UnlessNullOrWhitespace(_localize);
+    }
+    
+    /// <summary>
+    /// Resolves a description method by calling the specified static method.
+    /// </summary>
+    /// <param name="argument">The argument that has the description method attribute</param>
+    /// <param name="methodName">The name of the method to call</param>
+    /// <returns>The description returned by the method, or null if the method cannot be resolved</returns>
+    protected virtual string? ResolveDescriptionMethod<T>(T argument, string methodName) where T : IArgument
+    {
+        try
+        {
+            // Get the declaring type from the argument's custom attributes
+            Type? declaringType = null;
+            
+            if (argument.CustomAttributes is ParameterInfo paramInfo)
+            {
+                declaringType = paramInfo.Member.DeclaringType;
+            }
+            else if (argument.CustomAttributes is PropertyInfo propInfo)
+            {
+                declaringType = propInfo.DeclaringType;
+            }
+            
+            if (declaringType == null)
+            {
+                return $"Error: Could not determine declaring type for method '{methodName}'";
+            }
+            
+            // Find the method
+            var method = declaringType.GetMethod(methodName, 
+                BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic,
+                null, Type.EmptyTypes, null);
+                
+            if (method == null)
+            {
+                return $"Error: Method '{methodName}' not found in type '{declaringType.Name}'";
```

Given this is returning a string, will the string be included in the help output the end user sees?  I would expect this to throw an InvalidConfigurationException at some point so devs are warned early when there is an error.  

Although, now that I say that, if this is only discovered when help is run for a command, then it's all the more likely a misconfiguration will escape detection.  Perhaps we should resolve the method (but not execute it) in the DefinitionMappingExtensions, setting it in a new `argument.DescriptionMethod` property. The other benefit is that we'll no longer need the "__DESCRIPTION_METHOD__" magic string.  You'll be able to check if DescriptionMethod is null.

Once that's done, you could hide the complexity of this behind argument.  the argument.Description property could return an internal field value or execute the DescriptionMethod if it exists.  Any place that calls argument.Description could take advantage of the feature.